### ISSUE TITLE
MultiResolutionDetector returns finfo.min/2 sentinel responses and unrelated LAFs when a level has fewer maxima than requested

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4089]
`MultiResolutionDetector.detect_features_on_single_level` fills the requested top-K with a
`torch.finfo(dtype).min / 2` sentinel whenever a pyramid level has fewer above-threshold NMS
maxima than its quota. Those sentinel rows are returned to the caller as ordinary responses,
paired with whatever LAFs `topk` happened to pick, so `KeyNetDetector` and any other
`MultiResolutionDetector` subclass can hand back keypoints that were never detected.

## Reproduction

```python
import torch
from kornia.feature import MultiResolutionDetector
from kornia.feature.responses import BlobHessian

torch.manual_seed(0)
det = MultiResolutionDetector(BlobHessian(), num_features=100)
lafs, resp = det(torch.rand(1, 1, 64, 64))

print("n sentinel:", int((resp < -1e30).sum()), "of", resp.numel())
print("min:", resp.min().item())
```

```
n sentinel: 70 of 100
min: -1.7014117331926443e+38
```

70 of the 100 returned features are padding. Their LAFs are real coordinates — `topk` returns
the first `k` flat indices once every candidate carries the same `fill` value — so nothing
downstream can tell them apart except by the response magnitude.

`KeyNetDetector(pretrained=False, num_features=100)` on the same image returns 100/100
sentinels, because the untrained `last_conv` + `ReLU` produce an all-zero response map and
every position is masked.

## Cause

`kornia/feature/scale_space_detector.py:504-507`

```python
fill = torch.finfo(det_flat.dtype).min / 2
det_masked = det_flat.masked_fill(det_flat <= self.score_threshold, fill)
k = min(num_kp, det_flat.numel())
top_scores, top_flat_idx = torch.topk(det_masked, k=k)
```

`k` is clamped against `det_flat.numel()` (the pixel count), not against the number of
positions that survived the mask, so `topk` keeps drawing from the masked-out set.

This is a regression from #3623. The previous implementation filtered explicitly:

```python
indices = indices[torch.where(scores_sorted > 0.0)]
yx = yx[:, indices[:num_kp]].t()
```

which returned a variable number of genuine maxima. (That version had its own defect —
`scores_sorted[:num_kp]` was returned unfiltered, so scores and LAFs could disagree in
length — so this is not a straight revert.)

## Impact

`MultiResolutionDetector.detect` only re-sorts when `lafs.shape[1] > self.num_features`, so on
the common path the sentinels survive to the caller. Anything that consumes the responses —
`LocalFeature`, score-weighted matching, a user threshold — sees `-1.7e38` rather than
"no detection". This is the same defect class as #4065.

## Suggested fix

Cap `k` at the number of positions above the threshold and pad the remainder to a neutral
score (0.0), or return the true count. Whatever shape contract is chosen, it needs to be
stated in the docstring: `MultiResolutionDetector` is documented as returning
`responses: shape [1xNx1]` with no mention that some rows may be padding.

Reproduced on `main` at 9abf06d9, torch 2.9.1, CPU.
